SDWebImage/SDWebImagePrefetcher.m

```diff
@@ -103,10 +106,12 @@ - (nullable SDWebImagePrefetchToken *)prefetchURLs:(nullable NSArray<NSURL *> *)
             // Current operation finished
             [sself callProgressBlockForToken:token imageURL:imageURL];
             
-            if (atomic_load_explicit(&(token->_finishedCount), memory_order_relaxed) + atomic_load_explicit(&(token->_skippedCount), memory_order_relaxed) >= atomic_load_explicit(&(token->_totalCount), memory_order_relaxed)) {
+            if (atomic_load_explicit(&(token->_finishedCount), memory_order_relaxed) == token->_totalCount) {
```

Comment:
Let's only make the `_finishedCount` to compare with `_totalCount`. `_skippedCount` is not necessary IMO.

Reply:
@zhongwuzw This should not change like this. Because `_finishedCount` means the task which is success. And `_skippedCount` means the task which is failed. Their sum should equal to `_totalCount` and call completion block. But not just `_finishedCount`

Reply:
@dreampiggy 🤔 Yeah, I have no context about these atomic variable. I just see you always make `_finishedCount` +1 no matter error and compare using `>=`.

Reply:
So the snippet code below is what you want? 🤔 

```
            if (error) {
                // Add last failed
                atomic_fetch_add_explicit(&(token->_skippedCount), 1, memory_order_relaxed);
            }
            else {
                atomic_fetch_add_explicit(&(token->_finishedCount), 1, memory_order_relaxed);
            }
            
            // Current operation finished
            [sself callProgressBlockForToken:token imageURL:imageURL];
            
            if (atomic_load_explicit(&(token->_finishedCount), memory_order_relaxed) + atomic_load_explicit(&(token->_skippedCount), memory_order_relaxed) == token->_totalCount) {
                // All finished
                if (!atomic_flag_test_and_set_explicit(&(token->_isAllFinished), memory_order_relaxed)) {
                    [sself callCompletionBlockForToken:token];
                    [sself removeRunningToken:token];
                }
            }
```

Reply:
Don't need to change the above. See the header files of `SDWebImagePrefetcher`.

The finished count is the count a task end, whether it's error or success. Always increase this number.
The skipped count is the count a task failed, only increase when error occur.
The total count is the number of input array url. This is not increased at all.

Why so that because you can `Cancel` a prefetch at any time. So the `finished count` may be smaller than `total count`.

I guess this may be a little wired, but SDWebImage 4.x already use this in all the prefetcher delegate. So we just keep this value meanings